lib/tasks/git.rake

```diff
@@ -0,0 +1,105 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+namespace :git do
+  desc 'Clean up your locale and remote repository'
+  task :clean do
+    # FIXME: change this to master once we are there
+    main_branch       = 'dev'
+    excluded_branches = [
+                          main_branch,
+                          'release/4.0',
+                          'release/4.1',
+                          'release/4.2'
+                        ].join('|')
+
+    # symbolic-ref gives us sth. like "refs/heads/foo" and we just need 'foo'
+    current_branch = `git symbolic-ref HEAD`.chomp.split('/').last
+
+    # we don't want to remove the current branch
+    excluded_branches << current_branch
+
+    if current_branch != main_branch
+      if $CHILD_STATUS.exitstatus == 0
+        puts "WARNING: You are on branch #{current_branch}, NOT #{main_branch}."
+      else
+        puts 'WARNING: You are not on a branch'
+      end
+      puts
+    end
+
+    puts 'Updating to most current code from origin ...'
+    `git fetch origin`
+
+    puts 'Pruning remote origin ...'
+    `git remote prune origin`
+
+    puts 'Fetching merged branches...'
+    remote_branches = `git branch -r --merged`
+                      .split("\n")
+                      .map(&:strip)
+                      .reject{ |b|
+                        !b.starts_with?('origin') ||
+                        excluded_branches.include?(b.split('/').drop(1).join('/'))
+                      }
+
+    local_branches = `git branch --merged`
+                     .gsub(/^\* /, '')
+                     .split("\n")
+                     .map(&:strip)
+                     .reject{ |b|
+                       !b.starts_with?('origin') ||
```

Comment:
For me `git branch --merged` will never output the prefix. I think you should only apply this condition for remote branches (`git branch -r --merged`).

Sample outputs from my console:

Local branches:
````
$ git branch --merged
  feature/aggregated_diffs
  feature/aggregated_mails
````

remote branches:
````
$ git branch -r --merged
  origin/dev
  origin/feature/19490-select2-3.5-upgrade
  origin/feature/aggregated_diffs
  origin/feature/aggregated_mails
````